internal/services/openai.go

```diff
@@ -69,12 +73,25 @@ func (o openAI) askChatGPT(body io.Reader) (*ChatGPTResponse, error) {
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Set("Authorization", "Bearer "+o.token)
 
+	var currentReqResponseTime time.Duration
+	trace := &httptrace.ClientTrace{
+		GotFirstResponseByte: func() {
+			currentReqResponseTime = time.Since(time.Now())
+		},
+	}
+	req = req.WithContext(httptrace.WithClientTrace(req.Context(), trace))
```

Comment:
Good to use a context here in a general, in case we need to cancel the long-running request.